src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java

```diff
@@ -384,7 +384,13 @@ protected Job asJob() {
 
 	        scheduledJob.scheduleBuild2(projectbuildDelay, action, new CauseAction(cause));
     	} else {
-	        LOGGER.log(Level.INFO, "trigger on merge request not set");
+            
+            if (triggerOnMergeRequest ) {
+	            LOGGER.log(Level.INFO, "trigger on merge request not set");
+            } else if ( this.isBranchAllowed(req.getObjectAttribute().getTargetBranch())) {
```

Comment:
Thanks, you're completely right..  I also didn't account for any potentially unknown cases since I didn't have an "else"...  I've put that back now.